.build/build-sonar.xml

```diff
@@ -0,0 +1,257 @@
+        <fail unless="env.SONAR_CASSANDRA_TOKEN" message="SONAR_CASSANDRA_TOKEN environment variable must be set when using Sonar server defined in env SONAR_SERVER_URL"/>
+        <property name="sonar.token" value="${env.SONAR_CASSANDRA_TOKEN}"/>
+    </target>
+
+    <target name="-maybe-setup-local-sonar-server" depends="sonar-init" unless="env.SONAR_SERVER_URL">
+        <property name="sonar.host.url" value="${sonar.host.url.default}"/>
+        <check-server-status url="${sonar.host.url}"/>
+
+        <exec executable="curl" failifexecutionfails="true" failonerror="true" logerror="true">
+            <arg line="-u admin:password"/>
+            <arg line="-X POST"/>
+            <arg line="-s"/>
+            <arg value="${sonar.host.url}/api/projects/create?name=${sonar.projectKey}&amp;mainBranch=${sonar.mainBranch}&amp;newCodeDefinitionType=PREVIOUS_VERSION&amp;project=${sonar.projectKey}&amp;visibility=public"/>
+        </exec>
+
+        <exec executable="curl" failifexecutionfails="true" failonerror="true" logerror="true">
+            <arg line="-u admin:password"/>
+            <arg line="-X POST"/>
+            <arg line="-s"/>
+            <arg value="${sonar.host.url}/api/qualityprofiles/restore"/>
+            <arg value="--form"/>
+            <arg value="backup=@${build.helpers.dir}/sonar-quality-profile.xml"/>
+        </exec>
+
+        <exec executable="curl" failifexecutionfails="true" failonerror="true" logerror="true">
+            <arg line="-u admin:password"/>
+            <arg line="-X POST"/>
+            <arg line="-s"/>
+            <arg value="${sonar.host.url}/api/qualityprofiles/add_project?language=java&amp;project=${sonar.projectKey}&amp;qualityProfile=cassandra"/>
+        </exec>
+
+        <exec executable="curl" failifexecutionfails="false" failonerror="true" logerror="false">
+            <arg line="-s"/>
+            <arg value="-u"/>
+            <arg value="admin:password"/>
+            <arg value="-X"/>
+            <arg value="POST"/>
+            <arg value="${sonar.host.url}/api/user_tokens/revoke?name=cassandra"/>
+        </exec>
+
+        <exec executable="curl" failifexecutionfails="true" failonerror="true" logerror="false">
+            <arg line="-s"/>
+            <arg value="-u"/>
+            <arg value="admin:password"/>
+            <arg value="-X"/>
+            <arg value="POST"/>
+            <arg value="${sonar.host.url}/api/user_tokens/generate?name=cassandra&amp;type=GLOBAL_ANALYSIS_TOKEN"/>
+            <redirector outputproperty="sonar.token">
+                <outputfilterchain>
+                    <tokenfilter>
+                        <replaceregex pattern=".*&quot;token&quot;:&quot;([^&quot;]+)&quot;.*" replace="\1"/>
+                    </tokenfilter>
+                </outputfilterchain>
+            </redirector>
+        </exec>
+    </target>
+
+    <target name="stop-sonar-server" depends="sonar-init">
```

Comment:
@jacek-lewandowski I would rename this target to "sonar-server-start", when I am in the console, and I type "ant son<tab>", it does not offer this target and then I need to figure out which one it is ... it is easier for a user if all targets are grouped under "sonar-something".

Reply:
@jacek-lewandowski also, if I do:

    ant start-sonar-server
    ant stop-sonar-server
    ant start-sonar-server

The last command fails, because there is still a container of name "sonarqube" but it is in exited state and it collides.

Would it be possible to invoke `docker rm sonarqube` upon `stop-sonar-server`?

Reply:
I deliberately didn't remove the container - the message at the end should says that. Until it is removed you can start it again and navigate the results. Also, I'd avoid removing the container because someone can already have some sonarqube container in the system and our script should not remove it

Reply:
OK, so ... do you think that "sonar-xyz" targets make sense?

Reply:
ah, that - yes, I've already changed that, though haven't committed yet